src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -793,26 +795,37 @@ void processFrame(Http2Frame frame) throws IOException {
                 }
                 return;
             }
-            if (frame instanceof PushPromiseFrame) {
-                PushPromiseFrame pp = (PushPromiseFrame)frame;
-                try {
-                    handlePushPromise(stream, pp);
-                } catch (UncheckedIOException e) {
-                    protocolError(ResetFrame.PROTOCOL_ERROR, e.getMessage());
-                    return;
-                }
-            } else if (frame instanceof HeaderFrame) {
-                // decode headers (or continuation)
-                try {
-                    decodeHeaders((HeaderFrame) frame, stream.rspHeadersConsumer());
-                } catch (UncheckedIOException e) {
-                    debug.log("Error decoding headers: " + e.getMessage(), e);
-                    protocolError(ResetFrame.PROTOCOL_ERROR, e.getMessage());
-                    return;
+
+            // While pus frame is not null, the only acceptable frame on this
+            // stream is a Continuation frame
+            if (pcs != null) {
+                if (frame instanceof ContinuationFrame cf) {
+                    handlePushContinuation(stream, cf);
```

Comment:
IOException should probably be caught and transformed in protocol error here to?
In case of protocol error when handling push promise (or their continuation) - should `pcs` be reset to null?
Maybe it doesn't matter since we're closing the connection anyway.

Reply:
In my opinion it would probably be safest to set pcs to null here yes.